OpenRA.Mods.Common/Traits/World/EditorActorLayer.cs

```diff
@@ -110,10 +110,11 @@ void ITickRender.TickRender(WorldRenderer wr, Actor self)
 		public virtual IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr)
 		{
 			if (wr.World.Type != WorldType.Editor)
-				return NoRenderables;
+				yield break;
```

Comment:
This is a pessimisation for allocation when the condition is false

```c#
if (false)
    return NoRenderables;
```

doesn't need to allocate, but 
```c#
if (false)
	yield break;
```

needs to allocate a compiler generated enumerable, which is empty when enumerated.

When we don't care about the lazy evaluating behaviour, the former is better.

Reply:
This only exists on editor world, so therte's no situation where the condition is false. Besides only one EditorActorLayer exists at a time

Reply:
Would it be worth moving the check to the ctor, and throw an exception instead?

Reply:
Sure, but it likely costs literally nothing